### PR TITLE
feat: redux toolkit added with add note functionality

### PR DESCRIPTION
**Pull request by vrx29**
Redux Toolkit Added
- Made Create new note 
- Display selected note

### CODE REVIEW

File: src/features/Notes/notesSlice.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  data: [] as Array<NotesType>,
+  currentNote: {} as NotesType
+};
+
+export const notesSlice = createSlice({
+  name: 'notes',
+  initialState,
+  reducers: {
+    addNote: (state, action: PayloadAction<NotesType>) => {
+      state.data.push(action.payload);
```
Comment (anik31) [inline src/features/Notes/notesSlice.tsx:19]: appreciation - liked that you've added typescript and redux

File: src/index.tsx [hunk 2]
```diff
@@ -4,12 +4,16 @@ import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
 import { BrowserRouter } from 'react-router-dom';
+import { store } from './app/store';
+import { Provider } from 'react-redux';
 
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
 root.render(
```
Comment (anik31) [inline src/index.tsx:11]: appreciation - good that you're using react-18 syntax

**Review by anik31 — APPROVED** [review]
Nice work, the code is very clean & readable. The functionality is also working fine in the preview.

File: src/App.tsx [hunk 3]
```diff
@@ -8,9 +9,12 @@ function App() {
     <>
```
Comment (Maianki) [inline src/App.tsx:9]: nitpick: fragments can be ignored.

**Review by Maianki — APPROVED** [review]
This looks great. Very neat and tidy code. Also, great folder structure loved it. 
You can go ahead with the merge.

**Pull request merged**